File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        area = 0
        lo = 0
        hi = n-1
        while(lo<hi):
            temp = (min(height[lo],height[hi])*(hi-lo))
            area = max(temp,area)
            if(height[lo]>height[hi]):
                hi-=1
            else:
                lo+=1
        return area

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length-1;
        int area = 0 ;
        int max_area=0;
        while(l<r){
            area = (Math.min(height[l],height[r]))*(r-l);
            max_area = Math.max(area,max_area);
            if (height[l]>height[r]){
                r--;
            }else{
                l++;
            }
        }
        return max_area;
    }
}